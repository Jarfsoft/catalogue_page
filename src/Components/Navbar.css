.navbar {
  display: flex;
  align-items: center;
  justify-content: space-between;
  font-size: 1rem;
  font-family: Cambria, Cochin, Georgia, Times, 'Times New Roman', serif;
  background-color: rgb(0, 162, 154);
}

.navbar img {
  width: 5rem;
}

.title {
  margin: 0.25rem 3rem;
}

.links {
  display: flex;
  height: 2rem;
  margin: 0 3rem;
}

.links p {
  margin: 0.5rem 0;
  font-weight: bolder;
  color: white;
}

.links select {
  margin: 0.5rem;
  height: 1.2rem;
  width: 7rem;
}

.btn-link {
  margin: 1rem 1rem;
  color: lightcoral;
}
