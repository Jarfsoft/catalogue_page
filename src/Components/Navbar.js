import { React } from 'react';
import { useDispatch } from 'react-redux';

import actions from '../Actions';
import { getPokemons } from '../pokeApi';
import logo from './pokemon.png';
import './Navbar.css';

const Navbar = () => {
  const dispatch = useDispatch();
  const setInfo = (list, type) => {
    const newState = { list, type };
    dispatch(actions.filter(newState));
  };

  const onFilterChange = (type) => {
    const newState = { list: [], type: '' };
    if (type !== '') getPokemons(1, type, setInfo);
    else dispatch(actions.filter(newState));
  };
  return (
    <main className="navbar">
      <div className="title">
        <img src={logo} alt="Logo" />
      </div>
      <div className="links">
        <p>Type filter: </p>
        <select name="typeFilter" required id="typeFilter" onChange={(e) => onFilterChange(e.target.value)}>
          <option value="">All</option>
          <option value="fire">Fire</option>
          <option value="water">Water</option>
          <option value="grass">Grass</option>
          <option value="electric">Electric</option>
          <option value="ice">Ice</option>
          <option value="rock">Rock</option>
          <option value="ground">Ground</option>
          <option value="bug">Bug</option>
          <option value="fighting">Fighting</option>
          <option value="ghost">Ghost</option>
          <option value="normal">Normal</option>
        </select>
      </div>
    </main>
  );
};

export default Navbar;
