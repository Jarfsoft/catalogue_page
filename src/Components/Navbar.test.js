import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Navbar from './Navbar';

describe('Navbar render', () => {
  const initialState = {
    list: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12],
    type: '',
  };
  const mockStore = configureStore();
  let store;

  describe('Snapshot check', () => {
    it('should render Navbar section', () => {
      store = mockStore(initialState);
      const navbar = renderer
        .create(<Provider store={store}><Navbar /></Provider>)
        .toJSON();
      expect(navbar).toMatchSnapshot();
    });
  });

  describe('Navbar contains', () => {
    it('should contain a select menu', () => {
      const navbar = render(<Provider store={store}><Navbar /></Provider>);
      const select = navbar.container.querySelector('select');
      expect(select).toHaveLength(12);
    });
    it('should contain a image', () => {
      const navbar = render(<Provider store={store}><Navbar /></Provider>);
      const select = navbar.container.querySelectorAll('img');
      expect(select).toHaveLength(1);
    });
  });
});
