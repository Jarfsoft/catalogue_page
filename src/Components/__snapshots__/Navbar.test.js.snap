// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar render Snapshot check should render Navbar section 1`] = `
<main
  className="navbar"
>
  <div
    className="title"
  >
    <img
      alt="Logo"
      src="pokemon.png"
    />
  </div>
  <div
    className="links"
  >
    <p>
      Type filter: 
    </p>
    <select
      id="typeFilter"
      name="typeFilter"
      onChange={[Function]}
      required={true}
    >
      <option
        value=""
      >
        All
      </option>
      <option
        value="fire"
      >
        Fire
      </option>
      <option
        value="water"
      >
        Water
      </option>
      <option
        value="grass"
      >
        Grass
      </option>
      <option
        value="electric"
      >
        Electric
      </option>
      <option
        value="ice"
      >
        Ice
      </option>
      <option
        value="rock"
      >
        Rock
      </option>
      <option
        value="ground"
      >
        Ground
      </option>
      <option
        value="bug"
      >
        Bug
      </option>
      <option
        value="fighting"
      >
        Fighting
      </option>
      <option
        value="ghost"
      >
        Ghost
      </option>
      <option
        value="normal"
      >
        Normal
      </option>
    </select>
  </div>
</main>
`;
